core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -982,6 +982,11 @@ public NativeImageInvokerInfo build() {
                         // --enable-monitoring=heapdump is not supported on Windows
                         monitoringOptions.add(NativeConfig.MonitoringOption.HEAPDUMP);
                     }
+
+                    if (graalVMVersion.compareTo(GraalVM.Version.VERSION_24_2_0) >= 0) {
+                        // Use built-in GraalVM thread dumper when possible
+                        monitoringOptions.add(NativeConfig.MonitoringOption.THREADDUMP);
+                    }
```

Comment:
@roberttoyonaga In https://github.com/quarkusio/quarkus/pull/48956#discussion_r2213310991 you said thread-dump is available since GraalVM for JDK 22 (which corresponds to internal version 24.0). Here you check for JDK 24 or later (24.2). Why?